monitor/api/tests/tests.py

```diff
@@ -0,0 +1,163 @@
+            test_invocations_ids = json.loads(invocations["ids"])
+            test_invocations_affected_rows = json.loads(invocations["affected_rows"])
+            test_invocations_statuses = json.loads(invocations["statuses"])
+            for index, invocation_id in enumerate(test_invocations_ids):
+                test_invocations.append(InvocationSchema(
+                    id=invocation_id,
+                    time_utc=test_invocations_times[index],
+                    status=test_invocations_statuses[index],
+                    affected_rows=self._parse_affected_row(affected_rows=test_invocations_affected_rows, index=index)
+                ))
+
+            totals = self._get_invocations_totals(test_invocations)
+            tests_invocations[sub_test_unique_id] = InvocationsSchems(
+                fail_rate=round(totals.errors / len(test_invocations), 2) if test_invocations else 0,
+                totals=totals,
+                invocations=test_invocations,
+                description=self._get_invocations_description(totals)
+            )
+        return tests_invocations
+    
+    @staticmethod
+    def _get_invocations_totals(invocations: List[InvocationSchema]) -> TotalsInvocationsSchema:
+        error_runs = len([invocation for invocation in invocations if invocation.status in ["error", "fail"]])
+        warrning_runs = len([invocation for invocation in invocations if invocation.status == "warn"])
+        passed_runs = len([invocation for invocation in invocations if invocation.status == "pass"])
+        return TotalsInvocationsSchema(
+            errors=error_runs,
+            warnings=warrning_runs,
+            passed=passed_runs,
+            resolve=0
+        )
+
+    @staticmethod
+    def _get_invocations_description(invocations_totals: TotalsInvocationsSchema) -> str:
+        all_invocations_count = invocations_totals.errors + invocations_totals.warnings + invocations_totals.passed + invocations_totals.resolve
+        return f"There were {invocations_totals.errors or 'no'} failures and {invocations_totals.warnings or 'no'} warnings out of the last {all_invocations_count} test runs."
+
+    @staticmethod
+    def _parse_affected_row(affected_rows: List[str], index: int) -> Optional[int]:
+        try:
+            return int(affected_rows[index])
+        except Exception:
+            return None
+
+    def get_total_tests_results(self, tests_metadata: Optional[List[TestMetadataSchema]] = None, days_back: Optional[int] = None):
+        tests: List[TestMetadataSchema] = tests_metadata if tests_metadata is not None else self.get_tests_metadata(days_back=days_back)
+        totals = dict()
+        for test in tests:
+            self._update_test_results_totals(
+                totals_dict=totals,
+                model_unique_id=test.model_unique_id,
+                status=test.status
+            )
+        return totals
+    
+    def get_total_tests_runs(
```

Comment:
cosmetics - maybe rename to get_totals_for_test_runs_screen?

Reply:
I don't like cupeling between BE and FE